### ISSUE TITLE
Need help understanding load balancing. App keeps getting "Too many connection" errors and increasing pgpool/postgresql replicas isn't allowing more than 105 connections

### ISSUE THREAD

**Issue by kennyatzeroeyes** [bitnami/charts#9805]
### Name and Version

bitnami/postgresql-ha 8.5.2

### What steps will reproduce the bug?

Kubernetes GKE cluster on node pool greater than 1 node
4 or greater pgpool replicas
4 or greater postgresql replicas
Enable external load balancer

Simple script to create connection array that doesn't close the connections and connects to the load balancer external IP
Increasing pgpool replicas seems to have no affect
Increasing postgresql replicas seems to have no affect

### Are you using any custom parameters or values?

Not that I'm aware

### What is the expected behavior?

I think we are confused with how to scale these connections. We initially thought increasing pgpools would indeed add to the max connections, but after some point of increasing pgpool replicas we can't go higher than 105 connections. We assume that this is because of max_connection setting which is default value of I believe to be 100 and that the other 5 connections are reserved superuser connections maybe? It seems the postgresql replicas share the same connections and so increasing postgresql replicas also has no affect. We searched for diagrams or documentation on how these pieces work together, but are not clear on whats happening behind the scenes

We also notice that when we connect to the external load balancer that it will select a pgpool based on what we believe to be CPU instead of available connections and therefor could select at random. Would like to understand if by design this is okay.

Our production app is suffering because we are hitting max connections and not sure how to mitigate it. Do we increase maxConnections in the helm chart values or another setting. If we do what is the purpose of replicating pgpool/postgresql in regard to connections.

Apologies if this is not the proper location for support. I created a new topic on the bitnami chart community also incase that is more proper to get clarity

We also see a lot of idle connections, but do not know if by design this is intended. Our application should be closing the connections upon completion, however we are using EF Core in our application which could obscure what is happening behind the scenes.

This issue is more to understand intended use for scaling/load balancing of the postgresql-ha chart

### What do you see instead?

No affect to maximum connections when we increase pgpool/postgresql replicas and inability to resolve "Too many connections" error.

### Additional information

Thank you in advance

**Comment by kennyatzeroeyes** [comment]
Example using python script to exhaust connections and not exhausting each pgpool/postgres, but instead load balancer seems to be serving based on another resource such as CPU. So even though there are likely more available connections we get errors because load balancer serves pgpool/postgresql which connections are exhausted. The number indicates total connections from querying "SELECT Count(*) FROM pg_stat_activity"

![image](https://user-images.githubusercontent.com/46905381/163451238-99ee8b0d-8b4d-4899-a4b1-ae34ce28a796.png)

**Comment by kennyatzeroeyes** [comment]
Attaching this incase its helpful showing the pods during the above test

![image](https://user-images.githubusercontent.com/46905381/163451597-b98114ea-8043-4376-9ef6-a4f0ce4639f1.png)

**Comment by kennyatzeroeyes** [comment]
Attaching this incase helpful showing the load balancer service that I'm connecting to

![image](https://user-images.githubusercontent.com/46905381/163451946-09ff33ba-5869-4bdf-99e1-97bbc5d4dedb.png)

**Comment by rafariossaa** [comment]
Hi,
  Depending on the load  you will need to tune the configuration. I found [this article](https://www.cybertec-postgresql.com/en/tuning-max_connections-in-postgresql/) and I think it could help you. 
  I would suggest to get a good understanding in how connections work in pgpool/postgresql before adding more nodes or changing the values.

**Comment by kennyatzeroeyes** [comment]
@rafariossaa thank you I will check out the article provided!

**Comment by kennyatzeroeyes** [comment]
@rafariossaa yeah this re-enforces my concern about adjust default settings blindly in postgresql/pgpool.conf so thank you was very helpful and exactly the kind of content to broaden our understanding we are hoping to find. 

We are setting pooling to true on the connection string
Our app is using EF Core and we can enable "Context Pooling" 

Question: So it sounds like we may just need to fine tune the pooling rather than adjusting max_connections? 
Question: Then also what is the point of increasing pgpool/postgresql replicas in the highly available helm chart in this case.. Specifically pgpool replicas as my understanding/assumption was that would scale the connections?

Thank you!

**Comment by rafariossaa** [comment]
Hi,
> Question: So it sounds like we may just need to fine tune the pooling rather than adjusting max_connections?
My suggestion is to get a good understanding to all the settings involved, you could also take a look to pgpool documentation.

> Question: Then also what is the point of increasing pgpool/postgresql replicas in the highly available helm chart in this case.. Specifically pgpool replicas as my understanding/assumption was that would scale the connections?
When you have more nodes you are more resilient to failures. It won't increase the connections available.
This numbers are very dependent on the user needs. So, they are left to the user to tune them accordingly to his needs.

**Comment by kennyatzeroeyes** [comment]
@rafariossaa again very helpful thank you. I will research more onto pgpool today and in the coming days

**Comment by rafariossaa** [comment]
Hi,
  @kennyatzeroeyes you are welcome. Don't hesitate to come back and share your findings. They could be useful for other users in a similar situation.

**Comment by kennyatzeroeyes** [comment]
@rafariossaa yeah I definately will. We decided to try a managed postgres and found the provider we used is scaling max_connections with the tier of their instances and the standard tier has 500 max connections scaling up to 1000. I still feel like we may have room for improvement on the application side with Entity Framework Core context pooling or pooling on the connection string. I still have on my plate to research more into pgpool as you had mentioned. I'll be sure to update anything I find out in regard to the self managed bitnami chart provided here.
